pkg/sources/git/git.go

```diff
@@ -466,78 +483,110 @@ func (s *Git) ScanCommits(ctx context.Context, repo *git.Repository, path string
 	// get the URL metadata for reporting (may be empty)
 	urlMetadata := getSafeRemoteURL(repo, "origin")
 
-	var depth int64
+	ctx = context.WithValues(ctx, "repo", urlMetadata)
 
 	gitDir := filepath.Join(path, gitDirName)
+	meta := scanMetadata{gitDir: gitDir, urlMetadata: urlMetadata, reporter: reporter, scanOptions: scanOptions}
+	s.startWorkers(ctx, meta, commitsWorker, commitChan)
 
-	logger := ctx.Logger().WithValues("repo", urlMetadata)
-	logger.V(1).Info("scanning repo", "base", scanOptions.BaseHash, "head", scanOptions.HeadHash)
-	for commit := range commitChan {
-		if len(scanOptions.BaseHash) > 0 {
-			if commit.Hash == scanOptions.BaseHash {
-				logger.V(1).Info("reached base commit", "commit", commit.Hash)
-				break
+	return nil
+}
+
+func (s *Git) startWorkers(ctx context.Context, meta scanMetadata, worker workerFunc, commitChan <-chan gitparse.Commit) {
+	var wg sync.WaitGroup
+
+	for i := 0; i < runtime.NumCPU(); i++ {
```

Comment:
Should this respect the concurrency flag?

Reply:
Ahh yes, I think that makes sense. Good call.